addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -28,6 +28,7 @@ class ChannelMember(models.Model):
     is_self = fields.Boolean(compute="_compute_is_self", search="_search_is_self")
     # channel
     channel_id = fields.Many2one("discuss.channel", "Channel", ondelete="cascade", required=True, auto_join=True)
+    is_owner = fields.Boolean("Is owner of the channel")
```

Comment:
Currently any user can self-promote themselves to owner.
As there is no specific restriction on this field, and members can still write themselves.

We should either enforce this field to be group protected (only to admins) and always read/write it as sudo.

Or should prevent members from writing their own member at all (and always write as sudo when writing owner).

Or we should detect with method overrides when attempting to self update owner and prevent it.

And create also might have the same problem in some situations.